backend/server.js

```diff
@@ -1,30 +1,147 @@
-import express from 'express'
-import bodyParser from 'body-parser'
-import cors from 'cors'
-import mongoose from 'mongoose'
-
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/authAPI"
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
-mongoose.Promise = Promise
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
-const port = process.env.PORT || 8080
-const app = express()
+import express from 'express';
+import cors from 'cors';
+import mongoose from 'mongoose';
+import crypto from 'crypto';
+import bcrypt from 'bcrypt';
+
+import netflixData from './data/netflix-titles.json';
+
+const mongoUrl = process.env.MONGO_URL || 'mongodb://localhost/authAPI';
+mongoose.connect(mongoUrl, {
+  useNewUrlParser: true,
+  useUnifiedTopology: true,
+  useCreateIndex: true,
+});
+// Added from Stack to remove deprication error. Don't know what it is.
+// mongoose.set('useCreateIndex', true);
+mongoose.Promise = Promise;
+
+const User = mongoose.model('User', {
+  email: {
+    type: String,
+    unique: true,
+    match: [/^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/, 'Please fill a valid email address']
+  },
+  password: {
+    type: String,
+    required: true,
+  },
+  accessToken: {
+    type: String,
+    default: () => crypto.randomBytes(128).toString('hex'),
+  },
+});
+
+const Movie = mongoose.model('Movie', {
+  title: String,
+  director: String,
+  release_year: Number,
+  description: String,
+});
+
+if (process.env.RESET_DB) {
+  const seedDB = async () => {
+    await Movie.deleteMany();
+
+    await netflixData.forEach((item) => {
+      const newMovie = new Movie(item);
+      newMovie.save();
+    });
+  };
+  seedDB();
+}
+
+const authenicateUser = async (req, res, next) => {
+  const accessToken = req.header('Authorization');
+  try {
+    const user = await User.findOne({ accessToken });
+    if (user) {
+      next();
+    } else {
+      res.status(401).json({ message: 'Not authenticated' });
+    }
+  } catch (error) {
+    res.status(400).json({ message: 'Invalid request', error });
+  }
+};
+
+const port = process.env.PORT || 8080;
+const app = express();
 
 // Add middlewares to enable cors and json body parsing
-app.use(cors())
-app.use(bodyParser.json())
+app.use(cors());
+app.use(express.json());
 
 // Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
-})
+  res.send('Hello world');
```

Comment:
Here you could add the listendpoints so we can see the endpoints when going to the API url :)